plugins/PrivacyManager/tests/Fixtures/MultipleSitesMultipleVisitsFixture.php

```diff
@@ -0,0 +1,498 @@
+    private static $searchCategory = array('', '', 'video', 'images', 'web', 'web');
+
+    public $dateTime = '2017-01-02 03:04:05';
+    public $trackingTime = '2017-01-02 03:04:05';
+    public $idSite = 1;
+    /**
+     * @var \PiwikTracker
+     */
+    private $tracker;
+
+    private $numSites = 5;
+    private $currentUserId;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->installLogTables();
+        $this->setUpWebsites();
+        $this->setUpLocation();
+        $this->trackVisitsForMultipleSites();
+    }
+
+    public function tearDown()
+    {
+        $this->tearDownLocation();
+    }
+
+    public function tearDownLocation()
+    {
+        LocationProvider::$providers = null;
+    }
+
+    public function setUpLocation()
+    {
+        $mock = new MockLocationProvider();
+        LocationProvider::$providers = array($mock);
+        LocationProvider::setCurrentProvider('mock_provider');
+        MockLocationProvider::$locations = array(
+            self::makeLocation('Stratford-upon-Avon', 'P3', 'gb', 123.456, 21.321), // template location
+
+            // same region, different city, same country
+            self::makeLocation('Nuneaton and Bedworth', 'P3', 'gb', $isp = 'comcast.net'),
+
+            // same region, city & country (different lat/long)
+            self::makeLocation('Stratford-upon-Avon', 'P3', 'gb', 124.456, 22.231, $isp = 'comcast.net'),
+
+            // same country, different region & city
+            self::makeLocation('London', 'H9', 'gb'),
+
+            // same country, different region, same city
+            self::makeLocation('Stratford-upon-Avon', 'G5', 'gb', $lat = null, $long = null, $isp = 'awesomeisp.com'),
+
+            // different country, diff region, same city
+            self::makeLocation('Stratford-upon-Avon', '66', 'ru'),
+
+            // different country, diff region (same as last), different city
+            self::makeLocation('Hluboká nad Vltavou', '66', 'ru'),
+
+            // different country, diff region (same as last), same city
+            self::makeLocation('Stratford-upon-Avon', '66', 'mk'),
+        );
+    }
+
+    public function installLogTables()
+    {
+        try {
+            $columns = DbHelper::getTableColumns(TestLogFoo::TABLE);
+            if (!empty($columns)) {
+                return; // already installed
+            }
+        } catch (\Exception $e) {
+            // not installed yet
+        }
+        $extraLogTables = array(new TestLogFooBar(), new TestLogFoo(), new TestLogFooBarBaz());
+        foreach ($extraLogTables as $extraLogTable) {
+            $extraLogTable->install();
+        }
+
+        Piwik::addAction('LogTables.addLogTables', function (&$logTables) use ($extraLogTables) {
+            foreach ($extraLogTables as $extraLogTable) {
+                $logTables[] = $extraLogTable;
+            }
+        });
```

Comment:
Not needed, just a note: you can also put events in `provideContainerConfig()` via `'observers.global'` entry. Has the added benefit of being added during tracking & other sub processes (due to TestingEnvironmentManipulator).